fetch_response.go

```diff
@@ -183,6 +185,16 @@ func (b *FetchResponseBlock) decode(pd packetDecoder, version int16) (err error)
 		}
 	}
 
+	// Follows the Java implementation for calculating the records lag:
+	// https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java
+	if len(b.RecordsSet) > 0 {
+		logEndOffset := *b.LastRecordsBatchOffset - 1
+		partitionLag := logEndOffset - b.RecordsSet[len(b.RecordsSet)-1].RecordBatch.LastOffset()
+		if lagMetric != nil {
+			lagMetric.Update(partitionLag)
+		}
+	}
```

Comment:
I think it makes sense to pull the `lagMetric != nil` check up so we avoid doing working out for nothing. That is:
```suggestion
	if lagMetric != nil && len(b.RecordsSet) > 0 {
		logEndOffset := *b.LastRecordsBatchOffset - 1
		partitionLag := logEndOffset - b.RecordsSet[len(b.RecordsSet)-1].RecordBatch.LastOffset()
		lagMetric.Update(partitionLag)
	}
```